components/constellation/frame.rs

```diff
@@ -23,14 +25,8 @@ pub struct Frame {
     /// The frame id.
     pub id: FrameId,
 
-    /// The timestamp for the current session history entry
-    pub instant: Instant,
-
-    /// The pipeline for the current session history entry
-    pub pipeline_id: PipelineId,
-
-    /// The URL for the current session history entry
-    pub url: ServoUrl,
+    /// The current session history entry.
+    pub current: FrameState,
```

Comment:
Fair enough, Although this does become hard to maintain once we add more info on `FrameState`. Maybe we could have an info struct that contains all the frame state info other than the pipeline id?